### PR TITLE
[triton] Update triton

### PR DESCRIPTION
**Pull request by illera88**
**Describe the pull request**

- #### What does your PR fix?
Updates Triton to a newer code version

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
Same as before

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

### CODE REVIEW

File: ports/triton/vcpkg.json [hunk 1]
```diff
@@ -27,13 +27,6 @@
     "z3"
   ],
   "features": {
-    "boost": {
-      "description": "Use Boost as multiprecision library",
-      "dependencies": [
-        "boost-multiprecision",
-        "boost-numeric-conversion"
-      ]
-    },
```
Comment (Adela0814) [inline ports/triton/vcpkg.json:36]: Please don't delete already existing features since it would break vcpkg upgrade. If the feature no longer works you can change the description to 'deprecated'.

File: ports/triton/portfile.cmake [hunk 2]
```diff
@@ -31,11 +27,11 @@ vcpkg_cmake_configure(
     OPTIONS
         ${FEATURE_OPTIONS}
         -DZ3_INTERFACE=ON
-        -DTRITON_BOOST_INTERFACE=OFF
+        -DBOOST_INTERFACE=OFF
         -DBUILD_SHARED_LIBS=${DYNAMICLIB}
         -DMSVC_STATIC=${STATICCRT}
         -DBUILD_EXAMPLES=OFF
-        -DENABLE_TEST=OFF
+        -DENABLE_TEST=OFF
```
Comment (Adela0814) [inline ports/triton/portfile.cmake:34]: There seems to be an unexpected space here.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/triton/portfile.cmake (hunk 2 above)
Comment (illera88) [reply ports/triton/portfile.cmake]: fixed

File: ports/triton/vcpkg.json (hunk 1 above)
Comment (illera88) [reply ports/triton/vcpkg.json]: fixed

File: ports/triton/portfile.cmake [hunk 3]
```diff
@@ -15,7 +12,6 @@ string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "static" STATICCRT)
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
         "python" PYTHON_BINDINGS
-        "boost"  BOOST_INTERFACE
```
Comment (Adela0814) [inline ports/triton/portfile.cmake:18]: Why did you delete this feature? I don't see it being deprecated upstream.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/triton/portfile.cmake (hunk 3 above)
Comment (illera88) [reply ports/triton/portfile.cmake:18]: [Here](https://github.com/JonathanSalwan/Triton/pull/1105) is the discussion about it. Using Boost as a library is an overkill when you can just use a header lib to do multiprecission math.

Boost has a lot of dependencies while the `wide-integer` built in Triton library does not.

Comment (Adela0814) [reply ports/triton/portfile.cmake:18]: I still recommend keeping the original. I noticed that you changed `-DBOOST_INTERFACE=OFF` to an option, I think this is better as a feature.

Comment (illera88) [reply ports/triton/portfile.cmake:18]: What should I change `-DBOOST_INTERFACE=OFF` for to get the feature from vcpkg command?

Comment (illera88) [reply ports/triton/portfile.cmake:18]: Just made a commit. Not sure if that should suffice.

Comment (Adela0814) [reply ports/triton/portfile.cmake:18]: When `boost` is a feature, it is equivalent to `-DBOOST_INTERFACE=OFF` when you install `triton` via `vcpkg install triton`, and it is equivalent to `-DBOOST_INTERFACE=ON` when you install `triton` via `vcpkg install triton[boost]`. You don't need to add it in options.
Also, you seem to be missing a space than the original.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by illera88** [comment]
Thanks for the clarification. It should be good to go now

File: ports/triton/portfile.cmake [hunk 4]
```diff
@@ -31,7 +28,7 @@ vcpkg_cmake_configure(
     OPTIONS
         ${FEATURE_OPTIONS}
         -DZ3_INTERFACE=ON
-        -DTRITON_BOOST_INTERFACE=OFF
+        -DBOOST_INTERFACE=${BOOST_INTERFACE}
```
Comment (dg0yt) [inline ports/triton/portfile.cmake:31]: `BOOST_INTERFACE` is already passed in `${FEATURE_OPTIONS}`. Please remove the explict parameter.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/triton/portfile.cmake (hunk 4 above)
Comment (illera88) [reply ports/triton/portfile.cmake]: you are right. Done!

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
@illera88 Thanks for the update!

@dg0yt Thanks for the review!
